packages/@aws-cdk/integ-tests-alpha/lib/assertions/providers/lambda-handler/types.ts

```diff
@@ -150,6 +150,14 @@ export interface AwsApiCallRequest {
    * @default - return all data
    */
   readonly outputPaths?: string[];
+
+  /**
+   * Types that can't be properly serialized with JSON.stringify()
+   * are marked here. For example, an entry "foo" -> "ArrayBufferView"
+   * signals to the decoder that "foo" should be handled separately
+   * and treated as an array of bytes.
+   */
+  readonly specialTypes?: Record<string, string>;
```

Comment:
This solution of marking certain parameters as having a different type cannot be applied recursively (or at least not conveniently).

Can we not do an object with a type tag, that we can apply recursively?

```ts
encode({
  string: 'string',
  obj: { field: 'value' },
  weirdarray: new Uint8Array([1, 2, 3]),
})
// => 
{
  string: 'string',
  obj: { field: 'value' },
  weirdarray: {
    '$type': 'ArrayBufferView',
    elements: [1, 2, 3],
  },
}
```